### ISSUE TITLE
ssh-agent-auth requires a ssh-key-file with bastion host

### ISSUE THREAD

**Issue by fpicot** [rancher/rke#1670]
When trying to use rke with a bastion host, and ssh-agent, it still requires that we provide an valid ssh_key_path. It seems to be the same issue as #809, except with a bastion host

**RKE version:**
0.2.7

**Docker version: (`docker version`,`docker info` preferred)**
```     Version:           18.09.1-ol
     API version:       1.39
     Go version:        go1.10.8
     Git commit:        e32a1bd
     Built:             Thu Jun  6 14:55:42 2019
     OS/Arch:           linux/amd64
     Experimental:      false
```
**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
OEL 7.6

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare-metal

**cluster.yml file:**
```
kubernetes_version: 'v1.15.2-rancher1-1'
ssh_agent_auth: true

bastion_host:
    address: <bastion address>
    user: rancher

nodes: 
<nodes list>
```
**Steps to Reproduce:**
With the  above cluster.yml, I get : 
```# rke up --config ./cluster.yml
INFO[0000] Initiating Kubernetes cluster
FATA[0000] Error while reading SSH key file: open /root/.ssh/id_rsa: no such file or directory
```
If if comment out the bastion_host section:
```# rke up --config ./cluster.yml
INFO[0000] Initiating Kubernetes cluster
INFO[0000] [certificates] Generating admin certificates and kubeconfig
INFO[0000] Successfully Deployed state file at [./cluster.rkestate]
```

**Results:**

**Comment by deniseschannon** [comment]
@alena1108 since this was fixed by the community, can you help cherry-pick it to the `release/v0.3` branch and create a forwardport issue?

**Comment by jiaqiluo** [comment]
The bug is reproduced with RKE v0.2.8
```
➜  rke --v
rke version v0.2.8
```

Steps:
- in my local machine, use `ssh-add` to add the private key to ssh-agent.
- use `ssh-add -l` to verify the key is added 
- rename the private key and public key files
- use `ssh_agent_auth` and `bastion_host` in the rancher-cluster.yml file  
```
ssh_agent_auth: true
bastion_host:
    address: [public-ip]
    user: ubuntu
    port: 22
nodes:
  - address: [private-ip]
    user: ubuntu
    role: [etcd, controlplane, worker]   
services:
  etcd:
    snapshot: true
    creation: 6h
    retention: 24h
  kube-api:
    pod_security_policy: false
```

- user `rke up ` to provision the cluster

Result:
the following error shows
```
INFO[0000] Initiating Kubernetes cluster
FATA[0000] Error while reading SSH key file: open /Users/jiaqi/.ssh/id_rsa: no such file or directory
```

**Comment by deniseschannon** [comment]
Available with RKE CLI v1.1.0-rc1

**Comment by jiaqiluo** [comment]
The bug fix is validated with rke CLI v1.1.0-rc2

```
./rke-v1.1.0-rc2 -v
rke version v1.1.0-rc2
```

the steps are the same as my previous comment

Results:
- now RKE is using the ssh-agent to connect to both the bastion server and nodes. The logs show it can connect to the nodes and start pulling images

```
./rke-v1.1.0-rc2 up --config rancher-cluster.yml
WARN[0000] This is not an officially supported version (v1.1.0-rc2) of RKE. Please download the latest official release at https://github.com/rancher/rke/releases/latest
INFO[0000] Running RKE version: v1.1.0-rc2
INFO[0000] Initiating Kubernetes cluster
INFO[0000] [certificates] Generating admin certificates and kubeconfig
INFO[0000] Successfully Deployed state file at [./rancher-cluster.rkestate]
INFO[0000] Building Kubernetes cluster
INFO[0000] [dialer] Setup tunnel for host [172.31.28.226]
INFO[0001] [network] Deploying port listener containers
INFO[0001] Pulling image [rancher/rke-tools:v0.1.51] on host [172.31.28.226], try #1
```
